optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/inverserelation/InverseRelationShadowVariableDescriptor.java

```diff
@@ -62,6 +62,22 @@ public void linkVariableDescriptors(DescriptorPolicy descriptorPolicy) {
         linkShadowSources(descriptorPolicy);
     }
 
+    /**
+     * The following table describes which combinations of source and shadow variable types are supported.
+     * The unsupported combinations are detected and fail fast.
+     *
+     * <pre>
+     * {@code
+     *          | singleton | collection
+     * -----------------------------------
+     *   basic  |     ✕     |     ✓
+     * chained  |     ✓     |     ✕
+     *    list  |     ✓     |     ✕
+     *}
+     * </pre>
```

Comment:
(nitpick)
In the table its not clear to me if the rows or columns are the "source" side.

It also seems elaborate. How about this instead:
```suggestion
     * Sourced on a basic genuine planning variable, the shadow type is a Collection (such as List or Set).
     * Sourced on a list or chained planning variable, the shadow variable type is a single instance.
```